packages/node/test/dev.test.ts

```diff
@@ -0,0 +1,81 @@
+import { forkDevServer, readMessage } from '../src/fork-dev-server';
+import { resolve } from 'path';
+import fetch from 'node-fetch';
+
+test('runs a mjs endpoint', async () => {
+  const child = forkDevServer({
+    maybeTranspile: true,
+    config: {},
+    isEsm: true,
+    isTypeScript: false,
+    meta: {},
+    require_: require,
+    tsConfig: undefined,
+    workPath: resolve(__dirname, './dev-fixtures'),
+    entrypoint: './esm-module.mjs',
+    devServerPath: resolve(__dirname, '../dist/dev-server.js'),
+  });
```

Comment:
**suggestion (non-blocking):** extract function

I think we could make it clearer what's being tested if we extracted this like:

```
function testForkDevServer(entrypoint: string) {
  const ext = extname(entrypoint);
  const isTypeScript = ext === '.ts';
  const isEsm = ext === '.mjs';
  return forkDevServer({
    maybeTranspile: true,
    config: {},
    isEsm,
    isTypeScript,
    meta: {},
    require_: require,
    tsConfig: undefined,
    workPath: resolve(__dirname, './dev-fixtures'),
    entrypoint,
    devServerPath: resolve(__dirname, '../dist/dev-server.js'),
  });
}
```

then calling it here would look like:

```
const child = testForkDevServer('./esm-module.mjs');
```

What do you think?

Reply:
Looks good hope I have time tomorrow to do that

Reply:
PR: https://github.com/vercel/vercel/pull/9509